arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/Parsing.kt

```diff
@@ -178,90 +178,73 @@ fun parseAnimationCurve(value: String): AnimationCurve {
-    return when (this) {
-        BasemapStyle.ArcGISImagery -> "arcgisImagery"
-        BasemapStyle.ArcGISImageryStandard -> "arcgisImageryStandard"
-        BasemapStyle.ArcGISImageryLabels -> "arcgisImageryLabels"
-        BasemapStyle.ArcGISLightGray -> "arcgisLightGray"
-        BasemapStyle.ArcGISLightGray -> null
-        BasemapStyle.ArcGISLightGrayLabels -> null
-        BasemapStyle.ArcGISDarkGray -> "arcgisDarkGray"
-        BasemapStyle.ArcGISDarkGrayBase -> null
-        BasemapStyle.ArcGISDarkGrayLabels -> null
-        BasemapStyle.ArcGISNavigation -> "arcgisNavigation"
-        BasemapStyle.ArcGISNavigationNight -> "arcgisNavigationNight"
-        BasemapStyle.ArcGISStreets -> "arcgisStreets"
-        BasemapStyle.ArcGISStreetsNight -> "arcgisStreetsNight"
-        BasemapStyle.OsmStreetsRelief -> "arcgisStreetsRelief"
-        BasemapStyle.ArcGISTopographic -> "arcgisTopographic"
-        BasemapStyle.ArcGISOceans -> "arcgisOceans"
-        BasemapStyle.ArcGISOceansBase -> null
-        BasemapStyle.ArcGISOceansLabels -> null
-        BasemapStyle.ArcGISTerrain -> "arcgisTerrain"
-        BasemapStyle.ArcGISTerrainBase -> null
-        BasemapStyle.ArcGISTerrainDetail -> null
-        BasemapStyle.ArcGISCommunity -> "arcgisCommunity"
-        BasemapStyle.ArcGISChartedTerritory -> "arcgisChartedTerritory"
-        BasemapStyle.ArcGISColoredPencil -> "arcgisColoredPencil"
-        BasemapStyle.ArcGISNova -> "arcgisNova"
-        BasemapStyle.ArcGISModernAntique -> "arcgisModernAntique"
-        BasemapStyle.ArcGISMidcentury -> "arcgisMidcentury"
-        BasemapStyle.ArcGISNewspaper -> "arcgisNewspaper"
-        BasemapStyle.ArcGISHillshadeLight -> "arcgisHillshadeLight"
-        BasemapStyle.ArcGISHillshadeDark -> "arcgisHillshadeDark"
-        BasemapStyle.ArcGISStreetsReliefBase -> null
-        BasemapStyle.ArcGISTopographicBase -> null
-        BasemapStyle.ArcGISChartedTerritoryBase -> null
-        BasemapStyle.ArcGISModernAntiqueBase -> null
-        BasemapStyle.OsmStandard -> "osmStandard"
-        BasemapStyle.OsmStandardRelief -> "osmStandardRelief"
-        BasemapStyle.OsmStandardReliefBase -> null
-        BasemapStyle.OsmStreets -> "osmStreets"
-        BasemapStyle.OsmStreetsRelief -> "osmStreetsRelief"
-        BasemapStyle.OsmStreetsReliefBase -> null
-        BasemapStyle.OsmLightGray -> "osmLightGray"
-        BasemapStyle.OsmLightGrayBase -> null
-        BasemapStyle.OsmLightGrayLabels -> null
-        BasemapStyle.OsmDarkGray -> "osmDarkGray"
-        BasemapStyle.OsmDarkGrayBase -> null
-        BasemapStyle.OsmDarkGrayLabels -> null
-        BasemapStyle.OsmHybrid -> "hybrid"
-        else -> null
-    }
+private val basemapStyleMapping = mapOf(
+    BasemapStyle.ArcGISImagery to "arcgisImagery",
+    BasemapStyle.ArcGISImageryStandard to "arcgisImageryStandard",
+    BasemapStyle.ArcGISImageryLabels to "arcgisImageryLabels",
+    BasemapStyle.ArcGISLightGray to "arcgisLightGray",
```

Comment:
Previously we used a "when" expression with the intention that we catch every case. At some point we added an "else" case, which defeats the purpose, but it would be nice to have an exhaustive when expression of possible 👍🏽 
This is handy if library updates include new base map styles, since this is a compile time error 🤔